crates/cxx-qt-gen/src/generator/cpp/property/mod.rs

```diff
@@ -27,24 +27,49 @@ pub fn generate_cpp_properties(
     let qobject_ident = qobject_idents.name.cxx_unqualified();
 
     for property in properties {
-        // Cache the idents as they are used in multiple places
+        // Cache the idents and flags as they are used in multiple places
         let idents = QPropertyNames::from(property);
         let cxx_ty = syn_type_to_cpp_type(&property.ty, type_names)?;
 
+        let flags = &property.flags;
+
         generated.metaobjects.push(meta::generate(&idents, &cxx_ty));
-        generated
-            .methods
-            .push(getter::generate(&idents, &qobject_ident, &cxx_ty));
-        generated
-            .private_methods
-            .push(getter::generate_wrapper(&idents, &cxx_ty));
-        generated
-            .methods
-            .push(setter::generate(&idents, &qobject_ident, &cxx_ty));
-        generated
-            .private_methods
-            .push(setter::generate_wrapper(&idents, &cxx_ty));
-        signals.push(signal::generate(&idents, qobject_idents));
+
+        // None indicates no custom identifier was provided
+        if flags.read.is_none() {
+            generated
+                .methods
+                .push(getter::generate(&idents, &qobject_ident, &cxx_ty));
+            generated
+                .private_methods
+                .push(getter::generate_wrapper(&idents, &cxx_ty));
+        }
+
+        // Checking custom setter wasn't provided
+        if flags
+            .write
+            .clone()
+            .is_some_and(|ident_option| ident_option.is_none())
+        {
+            if let Some(setter) = setter::generate(&idents, &qobject_ident, &cxx_ty) {
+                generated.methods.push(setter)
+            }
+
+            if let Some(setter_wrapper) = setter::generate_wrapper(&idents, &cxx_ty) {
+                generated.private_methods.push(setter_wrapper)
+            }
+        }
+
+        // Checking custom notifier wasn't provided
+        if flags
+            .notify
+            .clone()
+            .is_some_and(|ident_option| ident_option.is_none())
```

Comment:
guess all this fun is where the tri-state enum thing instead of Option<Name> in QPropertyNames could help ? did you have any further ideas on that?

eg
```rust
enum QPropertyFlagState {
   None,
   Auto(Name), // with auto generated name
   Custom(Name), // with custom name
}
```

Then here you would remove the flag check and just do this?

```rust
            if let Some(notify) = signal::generate(&idents, qobject_idents) {
                signals.push(notify)
            }
```

Then have the `generate` function return Some only when it's `Auto(...)` and not None or Custom ?

Reply:
Was thinking it would be Option<State> as then encapsulates that read can be Auto or Custom, but not None, and write and notify would be optional, I think I've left a comment somewhere about places where this refactor is useful

Reply:
Changed